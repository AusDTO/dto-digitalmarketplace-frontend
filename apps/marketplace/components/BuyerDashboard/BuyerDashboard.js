import React from 'react'
import LinkList from '@gov.au/link-list/lib/js/react.js'
import { Switch, Route } from 'react-router-dom'
import { rootPath } from 'marketplace/routes'
import BuyerDashboardMyBriefs from './BuyerDashboardMyBriefs'
import BuyerDashboardTeamBriefs from './BuyerDashboardTeamBriefs'
import BuyerDashboardTeamOverview from './BuyerDashboardTeamOverview'
import styles from './BuyerDashboard.scss'

const BuyerDashboard = props =>
  <div className={styles.container}>
    <div className={`${styles.header} row`}>
      <div className="col-md-12 col-sm-12">
        <h1 className="uikit-display-5">Opportunities</h1>
        <div className="row">
          <div className={`${styles.alignRight} col-xs-12`}>
            <LinkList
              inline
              items={[
                {
                  link: `${rootPath}/buyer-dashboard`,
                  text: 'My Briefs',
                  onClick: e => {
                    e.preventDefault()
                    props.history.push(`${rootPath}/buyer-dashboard`)
                  }
                },
                {
                  link: `${rootPath}/buyer-dashboard/team-briefs`,
                  text: 'Team Briefs',
                  onClick: e => {
                    e.preventDefault()
                    props.history.push(`${rootPath}/buyer-dashboard/team-briefs`)
                  }
                },
                {
                  link: `${rootPath}/buyer-dashboard/team-overview`,
                  text: 'Team overview',
                  onClick: e => {
                    e.preventDefault()
                    props.history.push(`${rootPath}/buyer-dashboard/team-overview`)
                  }
                }
              ]}
            />
            <a className={`${styles.firstButton} uikit-btn`} href="#url">
              Create new brief
            </a>
          </div>
        </div>
      </div>
    </div>
    <article role="main">
      <Switch>
        <Route exact path={props.match.url} render={() => <BuyerDashboardMyBriefs {...props} />} />
        <Route exact path={`${props.match.url}/team-briefs`} render={() => <BuyerDashboardTeamBriefs {...props} />} />
        <Route
          exact
          path={`${props.match.url}/team-overview`}
          render={() => <BuyerDashboardTeamOverview {...props} />}
        />
      </Switch>
    </article>
  </div>

export default BuyerDashboard
