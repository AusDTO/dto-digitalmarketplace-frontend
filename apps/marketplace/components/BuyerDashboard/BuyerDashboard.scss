.container {
  margin-bottom: 100px;
}

.header {
  margin-bottom: 40px;
}

.headingRow {
  font-size: 17px;
  font-weight: bold;
  color: #202751;
  border-bottom: solid 3px #202751;
  padding-bottom: 7px;
}

.priceRow {
  padding-top: 20px;
  padding-bottom: 21px;
  border-bottom: solid 1px #777777;
  color: #313131;
}

.badge {
  border-radius: 3px;
  border: solid 2px #0f979b;
  color: #0f979b;
  margin-bottom: 5px;
  padding: 2px 5px;
  width: fit-content;
}

.badgeRed {
  border-color: #ff0000;
  color: #ff0000;
}

.badgeBlue {
  border-color: #224a60;
  color: #224a60;
}

.badgeGreen {
  border-color: #40ac78;
  color: #40ac78;
}

.badgeYellow {
  border-color: #f5bf47;
  color: #313131;
}

.badgeGrey {
  border-color: #d6d6d6;
  color: #313131;
}

.greyRow {
  background-color: #fafafa;
}

.alignCenter {
  text-align: center;
}

.alignRight {
  text-align: right;
}

.firstButton {
  margin-right: 10px;
}
