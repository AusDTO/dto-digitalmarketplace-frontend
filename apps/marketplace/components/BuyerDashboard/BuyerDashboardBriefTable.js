import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
import loadBuyerDashboard from 'marketplace/actions/buyerDashboardActions'
import { rootPath } from 'marketplace/routes'
import { hasPermission, mapLot } from 'marketplace/components/helpers'
import BuyerDashboardHelp from './BuyerDashboardHelp'
import styles from './BuyerDashboard.scss'

export class BuyerDashboardBriefTable extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: true,
      briefCount: {
        closed: 0,
        draft: 0,
        live: 0,
        withdrawn: 0
      },
      briefs: [],
      organisation: ''
    }
  }

  componentDidMount() {
    this.props.loadData(this.props.status).then(response => {
      const data = {
        briefs: response.data.briefs,
        briefCount: response.data.brief_counts,
        organisation: response.data.organisation
      }
      this.setState({
        ...data,
        loading: false
      })

      this.props.dashboardLoaded(data)
    })
  }

  getLinkedBriefTitle(item) {
    const { isPartOfTeam, isTeamLead, teams } = this.props

    let name = ''
    let url = ''
    let permissionToEdit = false
    if (item.status === 'draft') {
      if (
        hasPermission(isPartOfTeam, isTeamLead, teams, 'create_drafts') ||
        hasPermission(isPartOfTeam, isTeamLead, teams, 'publish_opportunities')
      ) {
        permissionToEdit = true
      }
    } else {
      permissionToEdit = true
    }
    switch (item.lot) {
      case 'rfx':
        url = `${rootPath}/brief/${item.id}/overview/rfx`
        name = 'Untitled seek proposals and quotes'
        break
      case 'training2':
        url = `${rootPath}/brief/${item.id}/overview/training2`
        name = 'Training (untitled)'
        break
      case 'atm':
        url = `${rootPath}/brief/${item.id}/overview/atm`
        name = 'Untitled ask the market'
        break
      case 'specialist':
        url = `${rootPath}/brief/${item.id}/overview/specialist`
        name = 'Untitled specialist'
        break
      case 'digital-outcome':
        url = `/buyers/frameworks/${item.framework}/requirements/${item.lot}/${item.id}`
        name = 'Untitled outcome'
        break
      case 'digital-professionals':
      case 'training':
        url = `${rootPath}/brief/${item.id}/overview`
        break
      default:
        url = ''
    }
    if (!permissionToEdit) {
      url = `${rootPath}/request-access/create_drafts`
    }

    return <a href={url}>{item.name || name}</a>
  }

  render() {
    if (this.state.loading) {
      return <LoadingIndicatorFullPage />
    }

    if (this.state.briefs.length === 0) {
      return (
        <div>
          <div className="row">
            <div className="col-xs-12">
              <span />
              <p>You don&apos;t have any briefs. Create a new request to get started.</p>
            </div>
          </div>
          <div className="row">
            <div className="col-xs-12">
              <BuyerDashboardHelp />
            </div>
          </div>
        </div>
      )
    }

    return (
      <div className="row">
        <div className="col-xs-12">
          <table className={`${styles.resultListing} col-xs-12`}>
            <thead>
              <tr className={styles.headingRow}>
                <th scope="col" className={styles.colId}>
                  ID
                </th>
                <th scope="col" className={styles.colName}>
                  Name
                </th>
                <th scope="col" className={styles.colOwner}>
                  Created by
                </th>
                {this.props.additionalColumns.headers.map(ac => ac)}
              </tr>
            </thead>
            <tbody>
              {this.state.briefs.map(item => (
                <tr key={`item.${item.id}`}>
                  <td className={styles.colId}>{item.id}</td>
                  <td className={styles.colName}>
                    {this.getLinkedBriefTitle(item)}
                    <br />
                    <span>{mapLot(item.lot)}</span>
                    <span className={styles.internalReference}>{item.internalReference}</span>
                  </td>
                  <td className={styles.colOwner}>
                    {item.creators.map(c => (
                      <React.Fragment key={c}>
                        {c}
                        <br />
                      </React.Fragment>
                    ))}
                  </td>
                  {this.props.additionalColumns.columns.map(ac => ac(item))}
                </tr>
              ))}
            </tbody>
          </table>
          <BuyerDashboardHelp />
        </div>
      </div>
    )
  }
}

BuyerDashboardBriefTable.defaultProps = {
  status: null,
  additionalColumns: {},
  dashboardLoaded: () => ({
    briefCount: {
      closed: 0,
      draft: 0,
      live: 0,
      withdrawn: 0
    },
    briefs: [],
    organisation: ''
  })
}

BuyerDashboardBriefTable.propTypes = {
  status: PropTypes.string,
  additionalColumns: PropTypes.object,
  dashboardLoaded: PropTypes.func
}

const mapStateToProps = state => ({
  teams: state.app.teams,
  isTeamLead: state.app.isTeamLead,
  isPartOfTeam: state.app.isPartOfTeam
})

const mapDispatchToProps = dispatch => ({
  loadData: status => dispatch(loadBuyerDashboard(status))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BuyerDashboardBriefTable)
