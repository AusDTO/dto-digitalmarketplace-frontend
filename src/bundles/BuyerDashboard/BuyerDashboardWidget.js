import React from 'react'
import { Provider } from 'react-redux'
import RegisterComponent from '../../RegisterComponent'

import createStore from './redux/create'

import Overview from './components/Overview'
import TeamView from './components/TeamView'

const BuyerDashboardWidget = (props) => {
  const store = createStore(props)

  return (
    <Provider store={store} >
      <TeamView />
    </Provider>
  )
}

export default new RegisterComponent({ 'buyer-dashboard': BuyerDashboardWidget })
