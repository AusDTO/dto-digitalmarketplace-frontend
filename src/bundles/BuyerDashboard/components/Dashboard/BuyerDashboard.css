

@media (max-width: 767px) {
  .team-view {
    margin-left: -15px;
  }
}

.team-view-heading {
  font-size: 20px;
  margin-bottom: 35px;
}

.team-view-username {
  font-size: 32px;
  font-weight: bold;
}

.team-view-username span {
  display: block;
}

@media (max-width: 767px) {
  .team-view-username {
    width: 65%;
  }
}

.team-view-username span button {
  margin: 10px 0px 50px 0px;
  float: right;
  padding: 7px;
  font-size: 15px;
  font-weight: normal;
  width: 25%;
}

@media (max-width: 767px) {
  .team-view-username span button {
    float: left;
    width: 300px
  }
}

.team-view-username span a{
  width: 320px;
  height: 50px;
  font-size: 17px;
  float: right;
  text-align: center;
  color: white
}

.team-view-teamname {
  margin-bottom: 0px;
  font-size: 17px;
  color: #6e6e6e;
  line-height: initial;
}

.team-view-brief-button {
  display: block
}


.heading {
  margin-top: 0;
  margin-bottom: 1rem;
}

.filters {
  margin: 1.5rem 0 0;
  width: 74.48861%
}

.filters hr {
  margin: 0;
}

.filters a {
  font-size: 1.25rem;
}

.filters span {
  font-size: 1.5rem !important;
}

@media (max-width: 767px) {
  .filters {
    margin-top: -1.5rem;
    width: 100%
  }

  .filters a {
    font-size: 0.875rem;
  }

  .filters span {
    font-size: 1.0rem !important;
  }

  .filter {
    padding-bottom: 10px !important
  }
}

/**
 * These !importants are ugly
 * But they exist to combat ui-kit's high specificity
 * on anchor tag.
 */
.filter {
  display: inline-block;
  font-size: 1.25rem;
  margin-right: 1rem;
  border-bottom: none !important;
  padding: 10px
}

.filter:hover {
  background-color: transparent !important;
  cursor: pointer;
}

.filter span {
  font-size: 1.65rem;
  display: inline-block;
  margin-right: 0.25rem;
}

.active-filter.filter {
  background-color: #def4f9 !important;
  border-bottom: solid 6px #17788d !important;
  font-weight: 700;
}

.fadeIn {
  transition: opacity 0.3s ease-in-out;
  opacity: 1;
}
.fadeOut {
  transition: opacity 0.3s ease-in-out;
  opacity: 0;
}

.summary-item-free-text {
  color: #313131;
  text-decoration: none;
  outline: none;
  border-bottom: solid 1px #5bcbe3;
}

.closed-brief-links {
  white-space: nowrap;
}

