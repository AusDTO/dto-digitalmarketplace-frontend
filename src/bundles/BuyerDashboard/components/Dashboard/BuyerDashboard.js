import React, {Component} from 'react'
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux';
import {withRouter, Link} from 'react-router-dom';
import DashboardBriefs from '../DashboardBriefs';
import TeamOverview from '../TeamOverview';
import BriefChoice from '../BriefChoice/BriefChoice';

import './BuyerDashboard.css'

import {actionCreators as actions} from '../../redux/modules/team';

class BuyerDashboard extends Component {

  constructor(props){
    super(props)
    this.state = {
      briefChoice: false
    }
  }

  toggleBriefChoice(){
    this.setState({
      briefChoice: !this.state.briefChoice
    })
  }

  onSelectView(value) {
    const {updateView} = this.props;
    updateView(value)
  }

  render() {
    const {team, location} = this.props;
    let pathname = location.pathname;
    let briefView = pathname.slice(pathname.lastIndexOf('/') + 1, pathname.length);
    briefView = (['teambriefs', 'teamoverview'].includes(briefView) ? briefView : 'mybriefs')
    
    return (
      <section>
      {(this.state.briefChoice ?
        <section>
          <article>
            <div>
                <BriefChoice handleToggle={() => this.toggleBriefChoice()}/>
            </div>
          </article>
        </section>
        :
        <section>
          <article styleName="team-view">
            <section styleName="team-view-heading">
              <div id="team-view-teamname" styleName="team-view-teamname">
                {team.teamName}
              </div>
              <div styleName="team-view-username">
              <span>
              <h1> Dashboard</h1>
                <button type="button" onClick={() => this.toggleBriefChoice()}>Start a new brief</button>
              </span>
              </div>
            </section>
            <article styleName="filters" className="row">
              <div className="row">
                <div className="col-xs-12 col-sm-12">
                  <Link to="/buyers/mybriefs"
                        onClick={() => this.onSelectView('mybriefs')}
                        styleName={`${briefView === 'mybriefs' ? 'active-filter' : ''} filter`}>
                    My briefs
                  </Link>
                  <Link to="/buyers/teambriefs"
                        onClick={() => this.onSelectView('teambriefs')}
                        styleName={`${briefView === 'teambriefs' ? 'active-filter' : ''} filter`}>
                    Team briefs
                  </Link>
                  <Link to="/buyers/teamoverview"
                        onClick={() => this.onSelectView('teamoverview')}
                        styleName={`${briefView === 'teamoverview' ? 'active-filter' : ''} filter`}>
                    Team overview
                  </Link>
                </div>

                <hr/>
              </div>
            </article>
            <article>
              {(briefView === 'mybriefs') && (
                <DashboardBriefs
                  teamBriefView={false}
                  briefs={team.briefs}
                />
              )}
              {(briefView === 'teambriefs') && (
                <DashboardBriefs
                  teamBriefView={true}
                  briefs={team.teamBriefs}
                />
              )}
              {(briefView === 'teamoverview') && (
                <TeamOverview {...team}/>
              )}
            </article>
          </article>
        </section>
      )}
      </section>
    )
  }
}


const mapStateToProps = (state, ownProps) => {
  const {team} = state;
  return {
    team,
    ...ownProps
  };
};

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators({...actions,}, dispatch),
  updateView: (value) => {
    dispatch(actions.updateView(value))
  }
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(BuyerDashboard));

