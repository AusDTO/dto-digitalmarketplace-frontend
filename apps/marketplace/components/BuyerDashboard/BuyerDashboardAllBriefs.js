import React from 'react'
import PropTypes from 'prop-types'
import { statusConvert } from 'marketplace/components/helpers'
import BuyerDashboardBriefTable from './BuyerDashboardBriefTable'
import styles from './BuyerDashboard.scss'

const BuyerDashboardAllBriefs = props => (
  <BuyerDashboardBriefTable
    additionalColumns={{
      headers: [
        <th scope="col" key={1} className={styles.colStatus}>
          Status
        </th>
      ],
      columns: [
        item => (
          <td key={item.id} className={styles.colStatus}>
            <div
              className={`${styles.badge}
                ${(item.status === 'withdrawn' && styles.badgeGrey) ||
                  (item.status === 'live' && styles.badgeBlue) ||
                  (item.status === 'closed' && styles.badgeYellow) ||
                  styles.badgeGrey}`}
            >
              {statusConvert(item.status)}
            </div>
          </td>
        )
      ]
    }}
    briefCountUpdated={bc => props.briefCountUpdated(bc)}
  />
)

BuyerDashboardAllBriefs.propTypes = {
  briefCountUpdated: PropTypes.func.isRequired
}

export default BuyerDashboardAllBriefs
