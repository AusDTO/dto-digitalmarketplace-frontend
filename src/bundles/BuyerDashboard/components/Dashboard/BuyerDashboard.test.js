// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { Router as MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import createMemoryHistory from 'history/createMemoryHistory';
import renderer from 'react-test-renderer';
import createStore from '../../redux/create';

import sampleState from './BuyerDashboard.json'
import BuyerDashboard from './BuyerDashboard';

test('BuyerDashboard renders without errors', () => {
  const store = createStore(Object.assign({}, { _serverContext: {} },  sampleState));
  const history = createMemoryHistory();
  const component = renderer.create(
    <MemoryRouter history={history}>
      <Provider store={store}>
        <BuyerDashboard />
      </Provider>
    </MemoryRouter>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('BuyerDashboard renders view filters', () => {
  const store = createStore(Object.assign({}, { _serverContext: {} },  sampleState));
  const history = createMemoryHistory();
  const component = mount(
    <MemoryRouter history={history}>
      <Provider store={store}>
        <BuyerDashboard />
      </Provider>
    </MemoryRouter>
  )

  expect(component.containsMatchingElement(<div>{sampleState.team.teamName}</div>)).toBeTruthy();
})