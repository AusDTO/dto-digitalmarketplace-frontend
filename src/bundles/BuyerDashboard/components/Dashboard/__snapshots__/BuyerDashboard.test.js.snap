// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyerDashboard renders without errors 1`] = `
<section>
  <article
    styleName="team-view"
  >
    <section
      styleName="team-view-heading"
    >
      <div
        styleName="team-view-teamname"
      >
        Digital Transformation Agency
      </div>
      <div
        styleName="team-view-username"
      >
        Matt Smith
      </div>
    </section>
    <article
      className="row"
      styleName="filters"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-12"
        >
          <a
            href="/buyers/mybriefs"
            onClick={[Function]}
            styleName="active-filter filter"
          >
            My briefs
          </a>
          <a
            href="/buyers/teambriefs"
            onClick={[Function]}
            styleName=" filter"
          >
            Team briefs
          </a>
          <a
            href="/buyers/teamoverview"
            onClick={[Function]}
            styleName=" filter"
          >
            Team overview
          </a>
        </div>
      </div>
      <hr />
    </article>
    <article>
      <div
        className="content-main"
      >
        <section>
          <h2>
            Draft briefs
          </h2>
          <table
            className="content-table summary-item-body"
          >
            <thead
              className="summary-item-field-headings-visible"
            >
              <tr>
                <th
                  className="summary-item-field-heading-first"
                >
                  Brief Overview
                </th>
                <th
                  className="summary-item-field-heading"
                >
                  Created
                </th>
                <th
                  className="summary-item-field-heading"
                >
                  Unanswered questions
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                className="summary-item-row"
              >
                <td
                  className="summary-item-field"
                >
                  <a
                    href="/buyers/frameworks/digital-service-professionals/requirements/digital-outcome/119"
                  >
                    Matts Brief
                  </a>
                </td>
                <td
                  className="summary-item-field"
                >
                  Thursday 4 October 2016
                </td>
                <td
                  className="summary-item-field"
                >
                  <div>
                    17
                     required
                  </div>
                  <div>
                    7
                     optional
                  </div>
                </td>
              </tr>
              <tr
                className="summary-item-row"
              >
                <td
                  className="summary-item-field"
                >
                  <a
                    href="/buyers/frameworks/digital-marketplace/requirements/digital-outcome/345"
                  >
                    test outcome
                  </a>
                </td>
                <td
                  className="summary-item-field"
                >
                  Thursday 4 April 2017
                </td>
                <td
                  className="summary-item-field"
                >
                  <div>
                    17
                     required
                  </div>
                  <div>
                    7
                     optional
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </section>
        <div>
          <p>
            <a
              href="/buyers/frameworks/digital-service-professionals/requirements/digital-professionals"
            >
              Create a new opportunity for a digital specialist
            </a>
          </p>
          <p>
            <a
              href="/buyers/frameworks/digital-service-professionals/requirements/digital-outcome/"
            >
              Create a new opportunity for an outcome
            </a>
          </p>
        </div>
        <section>
          <h2>
            Live briefs
          </h2>
          You don't have any live briefs.
        </section>
        <section>
          <h2>
            Closed briefs
          </h2>
          You don't have any closed briefs.
        </section>
        <div>
          <h3>
            Need a hand?
          </h3>
          <span>
            <a
              href="/buyers-guide"
            >
              Read the buyers guide 
            </a>
            or
            <a
              href="mailto:marketplace@digital.gov.au"
            >
               send us an email 
            </a>
            - We can help you write your brief.
          </span>
        </div>
      </div>
    </article>
  </article>
</section>
`;
